roofit/roofitcore/src/RooLinkedList.cxx

```diff
@@ -752,6 +752,22 @@ RooFIter RooLinkedList::fwdIterator() const {
   return RooFIter(std::move(iterImpl));
 }
 
+RooLinkedListIterImpl RooLinkedList::begin() const {
+  return {this, true};
+}
+
+RooLinkedListIterImpl RooLinkedList::end() const {
+  return {this, nullptr, true};
+}
+
+RooLinkedListIterImpl RooLinkedList::rbegin() const {
+  return {this, false};
+}
+
+RooLinkedListIterImpl RooLinkedList::rend() const {
+  return {this, nullptr, false};
+}
+
```

Comment:
This is a good idea, but please check this:
PyROOT needs (maybe needed) a typedef to the type that the iterator points to. If you include that typedef, it can automatically generate python loops.
You can:
- test if it already works
- if not, take inspiration from RooAbsCollection's typedef
- ask `@ etejedor` (not tagging him yet to not create noise)

Reply:
Good point! Fortunately, it already works. But your comment with the automatic pythonizations reminded me that I should have also added `size()` and `empty()`, so that we get `__len__` for free in pyROOT and also an interface that is closer to the STL.

Reply:
Here the Python test snippet I used:

```Python
import ROOT

obj1 = ROOT.TNamed("1", "1")
obj2 = ROOT.TNamed("2", "2")

ROOT.SetOwnership(obj1, False)
ROOT.SetOwnership(obj2, False)

roolist = ROOT.RooLinkedList()

roolist.Add(obj1)
roolist.Add(obj2)

roolist.Print("v")

print(len(roolist))

for obj in roolist:
    print(obj)
```

Reply:
Bravo, this looks like it should become a test. 😉